src/macros.jl

```diff
@@ -1014,7 +1015,7 @@ macro variable(args...)
             hasstart = true
             value = esc(ex.args[2])
         elseif kwarg == :basename
-            quotvarname = esc(ex.args[2])
+            basename = esc(ex.args[2])
```

Comment:
This sets basename to a `Symbol` but above when we do `basename = string(getname(var))`, it is a string and `constructvariable!` creates a string. Why is this working ? :-P

Reply:
`esc` applied to a constant has no effect, but `ex.args[2]` is not always a constant. If there are objects spliced into the string like `"var-$i"` then we get an expression object that needs to be escaped.